packages/vega3-extension/src/index.ts

```diff
@@ -84,10 +84,27 @@ class RenderedVega3 extends Widget implements IRenderMime.IRenderer {
           loader,
           actions: true
         };
-        return vegaEmbed(this.node as HTMLBaseElement, data, options).then((result) => {
-          // result contains the Vega spec and view
+        return vegaEmbed(this.node as HTMLBaseElement, data, options).then((result: any) => {
+          // Add png representation of vega chart to output
+          if (!model.data['image/png']) {
+            return result.view.toImageURL('png').then((imageData: any) => {
+              const data = { ...model.data, 'image/png': imageData.split(',')[1] };
+              model.setData({ data });
+            });
+          }
           return void 0;
-        }).catch(console.warn);
+        }).catch(error => {
+          // Add stderr message to output
+          const stderr = `Javascript Error: ${error.message}. This usually means there's a typo in your chart specification. See the JavaScript console for the full traceback.`;
+          const data = { 'application/vnd.jupyter.stderr': stderr };
+          model.setData({ data });
```

Comment:
Does this overwrite the malformed vega data? Is that the intended behavior? I might still want it in the document, even if it is wrong.

Reply:
It does overwrite it just because I observed that if it's still in there, the default renderer for the output will be RenderedVega vs. the stderr renderer, the result being a blank output.

Reply:
Ah, right. I suppose another option would be to change the default rank of `RenderedVega`, but this is probably a better solution.